source/Missions/Services/Network/LiteNetP2PClient.cs

```diff
@@ -228,17 +230,15 @@ public void OnPeerConnected(NetPeer peer)
 
         public void OnPeerDisconnected(NetPeer peer, DisconnectInfo disconnectInfo)
         {
-            if (PeerServer != null && peer != PeerServer)
+            if (PeerServer != peer)
             {
                 var peerDisconnectedEvent = new PeerDisconnected(peer, disconnectInfo);
                 messageBroker.Publish(this, peerDisconnectedEvent);
-                return;
             }
-            if (peer == PeerServer)
+            else
             {
-                ServerDisconnected serverDisconnected = new ServerDisconnected();
+                ServerDisconnected serverDisconnected = new ServerDisconnected(peer, disconnectInfo);
```

Comment:
Sending the peer here seems unneccessary unless you plan on using it later somehow